### ISSUE TITLE
mp4 video not playing in firefox and chrome from shared folder

### ISSUE THREAD

**Issue by diehummel** [owncloud/core#25780]
### Steps to reproduce
1. Share a older with mp4 video Files.
2. Access the shared folder and try view the file.
3. With Windows Internet Explorer the Files are played.
4. With Firefox and Chrome (windows and ubuntu) i receive "The Video could not be loaded. ... server or network failure .. format not supported"

In owncloud.log i get following message.

Exception: {"Exception":"Error","Message":"Call to undefined function split()","Code":0,"Trace":"#0 \/var\/www\/clients\/client1\/web1\/drive\/lib\/private\/legacy\/files.php(256): OC_Files::parseHttpRangeHeader('0-1850', 43546897)\n#1 \/var\/www\/clients\/client1\/web1\/drive\/lib\/private\/legacy\/files.php(119): OC_Files::getSingleFile(Object(OC\\Files\\View), '\/Mars Videos\/20...', 'mars_060816.mp4', Array)\n#2 \/var\/www\/clients\/client1\/web1\/drive\/apps\/files_sharing\/lib\/Controllers\/ShareController.php(503): OC_Files::get('\/Mars Videos\/20...', 'mars_060816.mp4', Array)\n#3 [internal function]: OCA\\Files_Sharing\\Controllers\\ShareController->downloadShare('xb0al2Fs29A7BiF', 'mars_060816.mp4', '\/2016', '')\n#4 \/var\/www\/clients\/client1\/web1\/drive\/lib\/private\/AppFramework\/Http\/Dispatcher.php(159): call_user_func_array(Array, Array)\n#5 \/var\/www\/clients\/client1\/web1\/drive\/lib\/private\/AppFramework\/Http\/Dispatcher.php(89): OC\\AppFramework\\Http\\Dispatcher->executeController(Object(OCA\\Files_Sharing\\Controllers\\ShareController), 'downloadShare')\n#6 \/var\/www\/clients\/client1\/web1\/drive\/lib\/private\/AppFramework\/App.php(110): OC\\AppFramework\\Http\\Dispatcher->dispatch(Object(OCA\\Files_Sharing\\Controllers\\ShareController), 'downloadShare')\n#7 \/var\/www\/clients\/client1\/web1\/drive\/lib\/public\/AppFramework\/App.php(135): OC\\AppFramework\\App::main('ShareController', 'downloadShare', Object(OC\\AppFramework\\DependencyInjection\\DIContainer))\n#8 \/var\/www\/clients\/client1\/web1\/drive\/core\/routes.php(140): OCP\\AppFramework\\App->dispatch('ShareController', 'downloadShare')\n#9 [internal function]: OC\\Route\\Router->{closure}(Array)\n#10 \/var\/www\/clients\/client1\/web1\/drive\/lib\/private\/Route\/Router.php(280): call_user_func(Object(Closure), Array)\n#11 \/var\/www\/clients\/client1\/web1\/drive\/lib\/base.php(891): OC\\Route\\Router->match('\/s\/xb0al2Fs29A7...')\n#12 \/var\/www\/clients\/client1\/web1\/drive\/index.php(39): OC::handleRequest()\n#13 {main}","File":"\/var\/www\/clients\/client1\/web1\/drive\/lib\/private\/legacy\/files.php","Line":195} 

No errors in HTTP log.

### Expected behaviour
Tell us what should happen

### Actual behaviour
Tell us what happens instead

### Server configuration
**Operating system**:
CENTOS 7
**Web server:**
Nginx 1.10.1-1.el7
**Database:**
mariadb 10.1.16-1
**PHP version:**
7.0.9
**ownCloud version:** (see ownCloud admin page)
9.1.0.15
**Updated from an older ownCloud or fresh install:**
updated. 8.2 > 9 > 9.1
**Where did you install ownCloud from:**
tarball
**Signing status (ownCloud 9.0 and above):**

```
Login as admin user into your ownCloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- core
	- EXTRA_FILE
		- error/401.html
		- error/403.html
		- error/400.html
		- error/500.html
		- error/405.html
		- error/502.html
		- error/404.html
		- error/503.html

Raw output
==========
Array
(
    [core] => Array
        (
            [EXTRA_FILE] => Array
                (
                    [error/401.html] => Array
                        (
                            [expected] => 
                            [current] => e6ba03a8ccd6c5c97b29c91ac2ac41e0aef0ac3be00f7888dfc4bedff2260972d76d28068ef3786c4989990883ffa8d96316ec6b1fb0399c4f6794a2fad77c6e
                        )

                    [error/403.html] => Array
                        (
                            [expected] => 
                            [current] => 2e6167149fe6d1d94d8f0e08ef0d567e087b567d530b6bf016b8d7c9d2d460171e24c8ef25fb23bd9e708a4e05877da8981e39d04a69dabc7f8f684f1f268495
                        )

                    [error/400.html] => Array
                        (
                            [expected] => 
                            [current] => 3569e8939479b680d332aed3e9a1f7d978069f13d1514f6fd1d3ec9b13c83d83ae0b4def5f21d53fb0e07d3c7adfb7ae07802d3096aa5c4af67acfcb7c2a694b
                        )

                    [error/500.html] => Array
                        (
                            [expected] => 
                            [current] => f48893930f6d159a01a4bb144f5c89fadaad694ebb73a39363a8348fdf865b2315cde059f490781783733283701f3056fd540577506715a3a2454f97934e0b8f
                        )

                    [error/405.html] => Array
                        (
                            [expected] => 
                            [current] => 364e115ff7eaaa5ed77381553fa48409637cbc129cc1ce33bf3389b8beca7d0cf3c9c829fc19469b592aa6f832969495c83ea3e766d1dbaff86410309c243771
                        )

                    [error/502.html] => Array
                        (
                            [expected] => 
                            [current] => e6e55ada4f0501a174b1b542816b3c6e3e373991056faf374867a778755fa2e0e34e1d57c0d758edf30e5df539e70685173ab17e1cdd9a228bf28d5daf307109
                        )

                    [error/404.html] => Array
                        (
                            [expected] => 
                            [current] => a9e36844697823a141aee250a1059d902d587e5801aa7d96bcc08d5a0f17de9dba59cd6c378f42b21e1d9780d58b1e18b3fa9bf0526d9c0727e1505ceb6005e2
                        )

                    [error/503.html] => Array
                        (
                            [expected] => 
                            [current] => e06fbb0c14eeafc5e247a264714de766c839756bdc4946042584c0c1dc9caba3f15e47be532ac72f59872ffccaabcd376ff99ddb3b7d4e6af4436768083bb158
                        )

                )

        )

)

**List of activated apps:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your ownCloud installation folder
```
Enabled:
  - activity: 2.3.2
  - dav: 0.2.5
  - federatedfilesharing: 0.3.0
  - files: 1.5.1
  - files_mv: 0.8.2
  - files_pdfviewer: 0.8.1
  - files_sharing: 0.10.0
  - files_texteditor: 2.1
  - files_trashbin: 0.9.0
  - files_versions: 1.3.0
  - files_videoplayer: 0.9.8
  - notifications: 0.3.0
  - updatenotification: 0.2.1
Disabled:
  - comments
  - encryption
  - external
  - federation
  - files_antivirus
  - files_external
  - firstrunwizard
  - gallery
  - hls_streamplayer
  - html5_videoplayer
  - provisioning_api
  - systemtags
  - templateeditor
  - user_external
  - user_ldap

**The content of config/config.php:**
<?php
$CONFIG = array (
  'instanceid' => 'ociuqt50bfmo',
  'passwordsalt' => 'q5817BMuoFQRYi/e3sw34m67jhPdfu',
  'secret' => 'tvt9Whyw2OkVGeccmg1Z/1BwxG5/MikzQXRqtxoXkx4sEC1V',
  'trusted_domains' => 
  array (
    0 => 'xxxxxx.at',
  ),
  'datadirectory' => '/var/www/clients/client1/web1/drive/data',
  'overwrite.cli.url' => 'https://xxxxx.at',
  'dbtype' => 'mysql',
  'version' => '9.1.0.15',
  'dbname' => 'cloud',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => '',
  'dbpassword' => '',
  'logtimezone' => 'Europe/Berlin',
  'installed' => true,
  'theme' => '',
  'tempdirectory' => '/tmp',
  'appstore.experimental.enabled' => true,
  'loglevel' => 0,
  'maintenance' => false,
  'integrity.check.disabled' => false,
  'singleuser' => false,
  'htaccess.RewriteBase' => '/',
  'memcache.local' => '\\OC\\Memcache\\Redis',
  'redis' => 
  array (
    'host' => 'localhost',
    'port' => 6379,
  ),
  'memcache.locking' => '\\OC\\Memcache\\Redis',
);

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your ownCloud installation folder

{
    "system": {
        "instanceid": "ociuqt50bfmo",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "drive.hummel-web.at"
        ],
        "datadirectory": "\/var\/www\/clients\/client1\/web1\/drive\/data",
        "overwrite.cli.url": "https:\/\/drive.hummel-web.at",
        "dbtype": "mysql",
        "version": "9.1.0.15",
        "dbname": "cloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "Europe\/Berlin",
        "installed": true,
        "theme": "",
        "tempdirectory": "\/tmp",
        "appstore.experimental.enabled": true,
        "loglevel": 0,
        "maintenance": false,
        "integrity.check.disabled": false,
        "singleuser": false,
        "htaccess.RewriteBase": "\/",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "localhost",
            "port": 6379
        },
        "memcache.locking": "\\OC\\Memcache\\Redis"
    }
}

or 

Insert your config.php content here
(Without the database password, passwordsalt and secret)
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
no
**Are you using encryption:** yes/no
no
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no
### Client configuration
**Browser:**
IE,Chrome,Firefox
**Operating system:**
Windows 8.1, Windows10, Ubuntu 16.04
#### Web server error log
```
Insert your webserver log here
```

#### ownCloud log (data/owncloud.log)
```
Warning 	filesystem 	Storage wrapper 'sharePermissions' was not registered via the 'OC_Filesystem - preSetup' hook which could cause potential problems. 	2016-08-12T08:28:25+02:00
Error 	index 	Exception: {"Exception":"Error","Message":"Call to undefined function split()","Code":0,"Trace":"#0 \/var\/www\/clients\/client1\/web1\/drive\/lib\/private\/legacy\/files.php(256): OC_Files::parseHttpRangeHeader('0-1850', 43546897)\n#1 \/var\/www\/clients\/client1\/web1\/drive\/lib\/private\/legacy\/files.php(119): OC_Files::getSingleFile(Object(OC\\Files\\View), '\/Mars Videos\/20...', 'mars_060816.mp4', Array)\n#2 \/var\/www\/clients\/client1\/web1\/drive\/apps\/files_sharing\/lib\/Controllers\/ShareController.php(503): OC_Files::get('\/Mars Videos\/20...', 'mars_060816.mp4', Array)\n#3 [internal function]: OCA\\Files_Sharing\\Controllers\\ShareController->downloadShare('xb0al2Fs29A7BiF', 'mars_060816.mp4', '\/2016', '')\n#4 \/var\/www\/clients\/client1\/web1\/drive\/lib\/private\/AppFramework\/Http\/Dispatcher.php(159): call_user_func_array(Array, Array)\n#5 \/var\/www\/clients\/client1\/web1\/drive\/lib\/private\/AppFramework\/Http\/Dispatcher.php(89): OC\\AppFramework\\Http\\Dispatcher->executeController(Object(OCA\\Files_Sharing\\Controllers\\ShareController), 'downloadShare')\n#6 \/var\/www\/clients\/client1\/web1\/drive\/lib\/private\/AppFramework\/App.php(110): OC\\AppFramework\\Http\\Dispatcher->dispatch(Object(OCA\\Files_Sharing\\Controllers\\ShareController), 'downloadShare')\n#7 \/var\/www\/clients\/client1\/web1\/drive\/lib\/public\/AppFramework\/App.php(135): OC\\AppFramework\\App::main('ShareController', 'downloadShare', Object(OC\\AppFramework\\DependencyInjection\\DIContainer))\n#8 \/var\/www\/clients\/client1\/web1\/drive\/core\/routes.php(140): OCP\\AppFramework\\App->dispatch('ShareController', 'downloadShare')\n#9 [internal function]: OC\\Route\\Router->{closure}(Array)\n#10 \/var\/www\/clients\/client1\/web1\/drive\/lib\/private\/Route\/Router.php(280): call_user_func(Object(Closure), Array)\n#11 \/var\/www\/clients\/client1\/web1\/drive\/lib\/base.php(891): OC\\Route\\Router->match('\/s\/xb0al2Fs29A7...')\n#12 \/var\/www\/clients\/client1\/web1\/drive\/index.php(39): OC::handleRequest()\n#13 {main}","File":"\/var\/www\/clients\/client1\/web1\/drive\/lib\/private\/legacy\/files.php","Line":195} 
```

#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log 
c) ...
```

**Comment by RealRancor** [comment]
@diehummel This is a duplicate of https://github.com/owncloud/core/issues/25483 which gets fixed with the upcoming 9.1.1 in https://github.com/owncloud/core/pull/25488

**Comment by diehummel** [comment]
ah. thanks. haven't found this.
